src/_nebari/stages/kubernetes_ingress/__init__.py

```diff
@@ -67,23 +67,23 @@ def check_ingress_dns(stage_outputs: Dict[str, Dict[str, Any]], disable_prompt:
     directory = "stages/04-kubernetes-ingress"
 
     ip_or_name = stage_outputs[directory]["load_balancer_address"]["value"]
-    ip = socket.gethostbyname(ip_or_name["hostname"] or ip_or_name["ip"])
+    _, _, ips = socket.gethostbyname_ex(ip_or_name["hostname"] or ip_or_name["ip"])
     domain_name = stage_outputs[directory]["domain"]
 
     def _attempt_dns_lookup(
-        domain_name, ip, num_attempts=NUM_ATTEMPTS, timeout=TIMEOUT
+        domain_name, ips, num_attempts=NUM_ATTEMPTS, timeout=TIMEOUT
     ):
         for i in range(num_attempts):
             try:
-                resolved_ip = socket.gethostbyname(domain_name)
-                if resolved_ip == ip:
+                _, _, resolved_ips = socket.gethostbyname_ex(domain_name)
+                if set(resolved_ips) == set(ips):
```

Comment:
I think that I actually liked your original implementation better here. Even though a CNAME is better if your LB has multiple ips, by checking if the resolved ip is in the list of ips it leaves the possibility that someone could just do an A record for a single ip.

If we are going to use this approach where we check all the ips, then the error message recommend a CNAME in the case of multiple IPs.

Reply:
Cool. I will change it back.

Reply:
Done